test_scripts/Defects/7_1/3556_1_GetListOfPermission.lua

```diff
@@ -0,0 +1,51 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/SmartDeviceLink/sdl_core/issues/3556
+---------------------------------------------------------------------------------------------------
+-- Description: Check that SDL sends response to SDL.GetListOfPermission with appId in case there are no specific
+--   permissions for the app
+--
+-- Preconditions:
+-- 1. Clean environment
+-- 2. PT does not contain group with user_consent_prompt in default section
+-- 3. SDL, HMI, Mobile session are started
+-- 4. App is registered
+-- 5. App is activated
+-- Steps:
+-- 1. HMI sends GetListOfPermissions request
+-- SDL does:
+--  - sends GetListOfPermissions response with empty allowedFunctions array list
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require("user_modules/sequences/actions")
+local json = require('modules/json')
+
+--[[ Test Configuration ]]
+runner.testSettings.restrictions.sdlBuildOptions = {{ extendedPolicy = { "EXTERNAL_PROPRIETARY" }}}
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Functions ]]
+local function getListOfPermissions()
+  local rid = common.getHMIConnection():SendRequest("SDL.GetListOfPermissions", { appID = common.getHMIAppId() })
+  common.getHMIConnection():ExpectResponse(rid, {
+    result = {
+      code = 0,
+      method = "SDL.GetListOfPermissions",
+      allowedFunctions = json.EMPTY_ARRAY,
+      externalConsentStatus = json.EMPTY_ARRAY
```

Comment:
Since `compareValues()` function in ATF which is used in all `Expect*` functions does not distinguish empty table `{}` and empty array `[]` I would propose to add custom `ValidIf` to check `allowedFunctions` and `externalConsentStatus` parameters.

However it might be better to update compareValues()` function. Though it could be tricky.